Backend/app/special-zone/src/Domain/Action/SpecialZone.hs

```diff
@@ -0,0 +1,130 @@
+{-
+ Copyright 2022-23, Juspay India Pvt Ltd
+
+ This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License
+
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+
+module Domain.Action.SpecialZone where
+
+import qualified Domain.Types.SpecialZone as Domain
+import Environment (ApiKey)
+import EulerHS.Prelude hiding (id, state)
+import Kernel.External.Maps (LatLong)
+import Kernel.Storage.Esqueleto (runTransaction)
+import Kernel.Types.APISuccess
+import Kernel.Types.Common
+import Kernel.Types.Error
+import Kernel.Types.Id
+import Kernel.Utils.Common
+import Storage.Queries.EntryExit
+import Storage.Queries.SpecialZone
+
+lookupSpecialZone :: EsqDBFlow m r => LatLong -> m [Domain.SpecialZone]
+lookupSpecialZone latLon = do
+  mbSpecialZone <- findSpecialZoneByLatLong latLon
+  case mbSpecialZone of
+    Nothing -> pure []
+    Just specialZone -> pure [specialZone]
```

Comment:
its `maybeToList`
By the way why do we need list here, if it contains not more than one element?

Reply:
This was done to make it compatible with the existing fp, yeah, can be changed now